rllib/algorithms/dreamerv3/dreamerv3.py

```diff
@@ -0,0 +1,653 @@
+        ...     training_ratio=tune.grid_search([256, 512, 1024])
+        ... )
+        >>> # Set the config object's env.
+        >>> config = config.environment(env="CartPole-v1")  # doctest: +SKIP
+        >>> # Use to_dict() to get the old-style python config dict
+        >>> # when running with tune.
+        >>> tune.Tuner(  # doctest: +SKIP
+        ...     "DreamerV3",
+        ...     run_config=air.RunConfig(stop={"episode_reward_mean": 200}),
+        ...     param_space=config.to_dict(),
+        ... ).fit()
+    """
+
+    def __init__(self, algo_class=None):
+        """Initializes a DreamerV3Config instance."""
+        super().__init__(algo_class=algo_class or DreamerV3)
+
+        # fmt: off
+        # __sphinx_doc_begin__
+
+        # DreamerV3 specific settings:
+        self.model_size = "XS"
+        self.training_ratio = 1024
+
+        self.replay_buffer_config = {
+            "type": "EpisodeReplayBuffer",
+            "capacity": int(1e6),
+        }
+
+        self.lr = None
+        self.world_model_lr = 1e-4
+        self.actor_lr = 3e-5
+        self.critic_lr = 3e-5
+        self.batch_size_B = 16
+        self.batch_length_T = 64
+        self.horizon_H = 15
+        self.gae_lambda = 0.95  # [1] eq. 7.
+        self.entropy_scale = 3e-4  # [1] eq. 11.
+        self.return_normalization_decay = 0.99  # [1] eq. 11 and 12.
+        self.train_critic = True
+        self.train_actor = True
+        self.intrinsic_rewards_scale = 0.1
+        self.world_model_grad_clip_by_global_norm = 1000.0
+        self.critic_grad_clip_by_global_norm = 100.0
+        self.actor_grad_clip_by_global_norm = 100.0
+
+        # Reporting.
+        self.report_individual_batch_item_stats = False
+        self.report_dream_data = False
+        self.report_images_and_videos = False
+        self.gc_frequency_train_steps = 100
+
+        # Override some of AlgorithmConfig's default values with DreamerV3-specific
+        # values.
+        self.framework_str = "tf2"
+        self.gamma = 0.997  # [1] eq. 7.
+        # Do not use! Set `batch_size_B` and `batch_length_T` instead.
+        self.train_batch_size = None
+        self.env_runner_cls = DreamerV3EnvRunner
+        self.num_rollout_workers = 0
+        self.rollout_fragment_length = 1
+        # Since we are using a gymnasium-based EnvRunner, we can utilitze its
+        # vectorization capabilities w/o suffering performance losses (as we would
+        # with RLlib's `RemoteVectorEnv`).
+        self.remote_worker_envs = True
+        # Dreamer only runs on the new API stack.
+        self._enable_learner_api = True
+        self._enable_rl_module_api = True
+        # __sphinx_doc_end__
+        # fmt: on
+
+    @override(AlgorithmConfig)
+    def training(
+        self,
+        *,
+        model_size: Optional[str] = NotProvided,
+        training_ratio: Optional[float] = NotProvided,
+        gc_frequency_train_steps: Optional[int] = NotProvided,
```

Comment:
again super nit (non-blocker at all): You have it here. but in the constructor it is defined under reporting group of attributes.